pkg/internal/util/cli/logger.go

```diff
@@ -98,29 +145,75 @@ func (l *Logger) Errorf(format string, args ...interface{}) {
 func (l *Logger) V(level log.Level) log.InfoLogger {
 	return infoLogger{
 		logger:  l,
-		enabled: level <= l.Verbosity,
+		level:   level,
+		enabled: level <= l.verbosity,
 	}
 }
 
+// infoLogger implements log.InfoLogger for Logger
 type infoLogger struct {
 	logger  *Logger
+	level   log.Level
 	enabled bool
 }
 
+// Enabled is part of the log.InfoLogger interface
 func (i infoLogger) Enabled() bool {
 	return i.enabled
 }
 
+// Info is part of the log.InfoLogger interface
 func (i infoLogger) Info(message string) {
 	if !i.enabled {
 		return
 	}
-	i.logger.print(message)
+	// for > 0, we are writing debug messages, include extra info
+	if i.level > 0 {
+		i.logger.debug(message)
+	} else {
+		i.logger.print(message)
+	}
 }
 
+// Infof is part of the log.InfoLogger interface
 func (i infoLogger) Infof(format string, args ...interface{}) {
 	if !i.enabled {
 		return
 	}
-	i.logger.printf(format, args...)
+	// for > 0, we are writing debug messages, include extra info
+	if i.level > 0 {
+		i.logger.debugf(format, args...)
+	} else {
+		i.logger.printf(format, args...)
+	}
+}
+
+// bufferPool is a type safe sync.Pool of *byte.Buffer, guaranteed to be Reset
```

Comment:
`sync.Pool` is a bit naive in that it requires all entries to be weighted evenly, which is not true for this case (despite the standard library including a very similar sample! there's an upstream bug about this ...) 

However, since we are only using this in a short-lived process (logging in the `kind` CLI commands), it's actually preferable to hold on to the larger allocations than churn on freeing / re-allocating them anyhow.

If we wanted to support long lived processes with potentially large log entries, then we'd want to be more clever.

Reply:
If that becomes the case it actually shouldn't be very hard to update this to handle that (EG the simplest solution is to assume mostly short entries under some fixed maximum length and not retain buffers over that length). But for now, YAGNI.

Reply:
added the simple solution from klog (drop > 256 byte buffers)